### PR TITLE
Adds anonymous pulls procedure for image registry

### PR DESCRIPTION
**Pull request by stevsmit**
Version(s):
4.16+

Issue:
https://issues.redhat.com/browse/OCPBUGS-33723

Link to docs preview:
https://76038--ocpdocs-pr.netlify.app/openshift-enterprise/latest/registry/accessing-the-registry.html#enabling-anonymous-pulls_accessing-the-registry

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/enabling-anonymous-pulls.adoc [hunk 1]
```diff
@@ -0,0 +1,58 @@
+// Module included in the following assemblies:
+//
+// * registry/accessing-the-registry.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="enabling-anonymous-pulls_{context}"]
+= Enabling anonymous pulls
+
+In some cases, administrators might want to enable anonymous pulls from the image registry, for example, when using CI/CD pipelines to streamline access to base images, or reduce authentication overhead.
```
Comment (rolfedh) [inline modules/enabling-anonymous-pulls.adoc:9]:
```suggestion
In some cases, administrators might want to enable anonymous pulls from the image registry, for example, when using CI/CD pipelines to streamline access to base images or reduce authentication overhead. 
```

File: modules/enabling-anonymous-pulls.adoc [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+In some cases, administrators might want to enable anonymous pulls from the image registry, for example, when using CI/CD pipelines to streamline access to base images, or reduce authentication overhead. 
+
+Previously, enabling anonymous pulls involved configuring the internal image registry to accept such requests by running the following command:
+
+[source,terminal]
+----
+$ oc adm policy add-role-to-group system:image-puller system:unauthenticated --namespace openshift
+----
+
+With {product-title} {product-version}, changes made to the API server and the `selfsubjectaccessreviews` resource now require administrators to add unauthenticated groups to the `self-access-review` Cluster Role. This is only required for new clusters installed from {product-title} {product-version}; clusters that are upgrading to {product-version} from a previous version can continue using the `oc adm policy add-role-to-group system:image-puller system:unauthenticated --namespace openshift` command.
```
Comment (rolfedh) [inline modules/enabling-anonymous-pulls.adoc:18]:
```suggestion
With {product-title} {product-version}, changes to the API server and the `selfsubjectaccessreviews` resource now require administrators to add unauthenticated groups to the `self-access-review` Cluster Role. This is only required for new clusters installed from {product-title} {product-version}; clusters that are upgrading to {product-version} from a previous version can continue using the `oc adm policy add-role-to-group system:image-puller system:unauthenticated --namespace openshift` command. 
```

File: modules/enabling-anonymous-pulls.adoc [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+Previously, enabling anonymous pulls involved configuring the internal image registry to accept such requests by running the following command:
+
+[source,terminal]
+----
+$ oc adm policy add-role-to-group system:image-puller system:unauthenticated --namespace openshift
+----
+
+With {product-title} {product-version}, changes made to the API server and the `selfsubjectaccessreviews` resource now require administrators to add unauthenticated groups to the `self-access-review` Cluster Role. This is only required for new clusters installed from {product-title} {product-version}; clusters that are upgrading to {product-version} from a previous version can continue using the `oc adm policy add-role-to-group system:image-puller system:unauthenticated --namespace openshift` command. 
+
+Depending on your {product-title} version, and whether you installed a new cluster from version 4.16 or later or upgraded to one of those versions from an earlier release, you can use one of the following commands to enable anonymous pulls from the internal image registry.
```
Comment (rolfedh) [inline modules/enabling-anonymous-pulls.adoc:20]:
```suggestion
Depending on your {product-title} version and whether you installed a new cluster from version 4.16 or later or upgraded to one of those versions from an earlier release, you can use one of the following commands to enable anonymous pulls from the internal image registry.
```

**Review by rolfedh — APPROVED** [review]
LGTM. Very nice work. I've offered some suggestions for minor nits.

File: modules/enabling-anonymous-pulls-upgraded-cluster.adoc [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+:_mod-docs-content-type: CONCEPT
+[id="enabling-anonymous-pulls-upgraded-cluster_{context}"]
+= Anonymous pulls for upgraded clusters
+
+Use the following procedure to enable anonymous pulls if you have upgraded to {product-title} 4.16 or later from {product-title} 4.15 or earlier.
+
+.Procedure 
+
+. Enter the following command to see if the `system:unauthenticated` group is included in the `subjects` parameter of your `ClusterRoleBinding` object:
```
Comment (flavianmissi) [inline modules/enabling-anonymous-pulls-upgraded-cluster.adoc:13]: The idea with adding this verification was to not have to distinguish between new and existing clusters - if the user always checks if `self-access-reviewers` have are available to `system:unauthenticated` then this alone can decide whether they need execute procedure "Create a file named `add-self-access-review-unauth.yaml` and add the following content" or not.
WDYT?

File: modules/enabling-anonymous-pulls-new-cluster.adoc [hunk 5]
```diff
@@ -0,0 +1,67 @@
+// Module included in the following assemblies:
+//
+// * registry/accessing-the-registry.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="enabling-anonymous-pulls-new-cluster_{context}"]
+= Anonymous pulls for new clusters
+
+Use the following procedure to enable anonymous pulls if you have installed a new cluster from {product-title} 4.16 or later.
```
Comment (flavianmissi) [inline modules/enabling-anonymous-pulls-new-cluster.adoc:9]: This procedure will not enable anonymous pulls to the registry. It only enables anonymous requests to the self access reviewer resource, which is required in new clusters in addition to the below step:
```
oc adm policy add-role-to-group system:image-puller system:unauthenticated --namespace <namespace>
```

File: modules/enabling-anonymous-pulls-upgraded-cluster.adoc [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: self-access-reviewer
+subjects:
+- apiGroup: rbac.authorization.k8s.io
+  kind: Group
+  name: system:authenticated
+----
+
+. If your cluster has the `self-access-reviewer` cluster role binding available to the `system:unauthenticated` group, anonymous pulls are already enabled for your cluster. You do not need to create a new cluster role binding, and can move on to the following step.
```
Comment (flavianmissi) [inline modules/enabling-anonymous-pulls-upgraded-cluster.adoc:43]:
```suggestion
. If your cluster has the `self-access-reviewer` cluster role binding available to the `system:unauthenticated` group, you do not need to create a new cluster role binding, and can move on to the following step.
```

File: modules/enabling-anonymous-pulls-upgraded-cluster.adoc [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+----
++
+.Example output
++
+[source,terminal]
+----
+clusterrolebinding.rbac.authorization.k8s.io/restore-self-access-reviewers-unauth created
+----
+
+. Enter the following command to create the new cluster role binding that enables anonymous requests to the self access reviewer source:
```
Comment (flavianmissi) [inline modules/enabling-anonymous-pulls-upgraded-cluster.adoc:74]: The wording here isn't quite right... `oc adm policy add-rol-to-group` doesn't really create a new cluster role binding right? maybe it's copy/pasta from the previous procedure?

File: modules/enabling-anonymous-pulls-upgraded-cluster.adoc [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
++
+[source,terminal]
+----
+$ oc adm policy add-role-to-group system:image-puller system:unauthenticated --namespace <namespace>
+----
+
+.Verification
```
Comment (flavianmissi) [inline modules/enabling-anonymous-pulls-upgraded-cluster.adoc:81]: QE uses `oc image info` to verify that anonymous pulls are enabled. What do you think of doing the same here?
You will find an example in this bug: https://issues.redhat.com/browse/OCPBUGS-33453

File: modules/enabling-anonymous-pulls.adoc [hunk 9]
```diff
@@ -0,0 +1,130 @@
+// Module included in the following assemblies:
+//
+// * registry/accessing-the-registry.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="enabling-anonymous-pulls_{context}"]
+= Enabling anonymous pulls
+
+The following procedure shows you how to check if your cluster has the proper roles and permissions for anonymous pulls and, if it does not, how to enable anonymous requests to the self access reviewer source.
```
Comment (flavianmissi) [inline modules/enabling-anonymous-pulls.adoc:9]: This sentence mixes up the enabling of anonymous pulls with anonymous requests to the self access reviewer resource - we should be consistent.
```suggestion
The following procedure shows you how to check if your cluster has the proper roles and permissions for anonymous pulls and, if it does not, how to enable anonymous pulls.
```

**Review by flavianmissi — COMMENTED** [review]
left a small comment, but looks good otherwise!

**Comment by flavianmissi** [comment]
we are discussing a potential code change that would avoid the need to create the new cluster role binding - let's wait and see how that progresses before merging this @stevsmit .

File: modules/anonymous-pulls.adoc [hunk 10]
```diff
@@ -0,0 +1,9 @@
+// Module included in the following assemblies:
+//
+// * registry/accessing-the-registry.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="anonymous-pulls_{context}"]
+= Anonymous pulls
+
+In some cases, administrators might want to enable anonymous pulls from the image registry, for example, when using CI/CD pipelines to streamline access to base images or reduce authentication overhead. To enable anonymous pulls, administrators must add unauthenticated groups to the `self-access-review` Cluster Role and grant the `system:image-puller` role to the `system:unauthenticated` group.
```
Comment (xiuwang) [inline modules/anonymous-pulls.adoc:9]:
```suggestion
In some cases, administrators might want to enable anonymous pulls from the image registry, for example, when using CI/CD pipelines to streamline access to base images or reduce authentication overhead. To enable anonymous pulls, administrators must add `system:unauthenticated` group to the `self-access-reviewer` Cluster Role and grant the `system:image-puller` role to the `system:unauthenticated` group.
```

File: modules/enabling-anonymous-pulls.adoc [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+----
++
+.Example output
++
+[source,terminal]
+----
+clusterrolebinding.rbac.authorization.k8s.io/restore-self-access-reviewers-unauth created
+----
+
+. Enter the following command to grant the `system:image-puller` role to the `system:unauthenticated` group in the specified namespace, enabling anonymous users to pull images:
```
Comment (xiuwang) [inline modules/enabling-anonymous-pulls.adoc:74]: How about add openshift namespace example, to keep consistent with step in verification.
```suggestion
. Enter the following command to grant the `system:image-puller` role to the `system:unauthenticated` group in the specified namespace,such as openshift namespcae, enabling anonymous users to pull images:
```

File: modules/enabling-anonymous-pulls.adoc [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
++
+[source,terminal]
+----
+$ oc adm policy add-role-to-group system:image-puller system:unauthenticated --namespace <namespace>
```
Comment (xiuwang) [inline modules/enabling-anonymous-pulls.adoc:78]:
```suggestion
$ oc adm policy add-role-to-group system:image-puller system:unauthenticated --namespace openshift
```

File: modules/enabling-anonymous-pulls.adoc [hunk 13]
```diff
@@ -0,0 +1,130 @@
...
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: self-access-reviewer
+subjects:
+- apiGroup: rbac.authorization.k8s.io
+  kind: Group
+  name: system:authenticated
+----
+
+. If your cluster has the `self-access-reviewer` cluster role binding available to the `system:unauthenticated` group, you do not need to create a new cluster role binding, and can move on to the following step.
```
Comment (xiuwang) [inline modules/enabling-anonymous-pulls.adoc:43]: Maybe
```suggestion
. If your cluster has the cluster role binding that `self-access-reviewer` cluster role is binded to the `system:unauthenticated` group, you do not need to create a new cluster role binding, and can move on to the following step.
```

File: modules/enabling-anonymous-pulls.adoc [hunk 14]
```diff
@@ -0,0 +1,130 @@
...
++
+If `self-access-reviewer` is not available to the `system:unauthenticated` group, or the cluster role binding does not exist, you can create a file named `add-self-access-review-unauth.yaml` and apply it with the following command:
```
Comment (xiuwang) [inline modules/enabling-anonymous-pulls.adoc:45]: Maybe
```suggestion
If `self-access-reviewer` is not available to the `system:unauthenticated` group, you can create a file named `add-self-access-review-unauth.yaml` and apply it with the following command:
```

**Comment by xiuwang** [comment]
I give some suggestions for minor nits, other looks good to me.

**Pull request closed without merging**